examples/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+# (c) 2022 Copyright, Real-Time Innovations, Inc. All rights reserved.
+# No duplications, whole or partial, manual or electronic, may be made
+# without express written permission.  Any such copies, or revisions thereof,
+# must display this notice unaltered.
+# This code contains trade secrets of Real-Time Innovations, Inc.
+cmake_minimum_required(VERSION 3.11)
+project(rtiexamples)
+
```

Comment:
We should add options to enable and disable building examples for some of the components of RTI Connext. At a minimum, I'd include the following options via CMake `option()`:

- `CONNEXTDDS_BUILD_PERSISTENCE_SERVICE_EXAMPLES`
- `CONNEXTDDS_BUILD_RECORDING_SERVICE_EXAMPLES`
- `CONNEXTDDS_BUILD_ROUTING_SERVICE_EXAMPLES`
- `CONNEXTDDS_BUILD_SECURITY_EXAMPLES`

Based on those options you can skip the compilation of those examples. We do something to disable the build of some examples based on certain CMake configurations

Reply:
I think we should basically make any example that can't be built out of the box (like security examples or the mongo db example). We should make an inventory of which examples should be optional to be built.